design/ErrorConventions.md

```diff
@@ -54,26 +54,39 @@ passed into `o.f`.
     * This is explicitly made more strict than the usual WebIDL dictionary
       binding rules.
 
-* If any object in `A` is missing a required key (given current features): Error **must** be synchronous.
+* If any object in `A` contains any key that is not known to the browser (is not
+  present in its WebIDL code): No error.
+    * This is how WebIDL works.
+    * If possible browsers could choose to issue warnings for unrecognized keys.
+
+* If any object in `A` sets any key that is known to the browser, but part of a
+  non-enabled feature, is set to any value other than its default:
+  Error **must not** be synchronous.
```

Comment:
If we have 2 browsers: one supporting feature X and one not supporting feature X, and we have an application that does *not* enable feature X but uses some of its types or enum values. Wouldn't we want the behavior of this application to match? Or rather, wouldn't we at least *permit* the behavior to match?
Right now this sounds like the behavior has to be different.

Reply:
It would be nice if it matched, yes, but specifically in the case of dictionary members added by features, we can't make it match unless we make every new dictionary member have type `any`. As we previously discussed, that doesn't seem worth doing.

So since WebIDL is already going to be strict about the type of the member, we may as well be slightly stricter about the value of the member, since it will provide better errors.